docs/discussion/lwip_ipv6.md

```diff
@@ -0,0 +1,139 @@
+can be added to the CHIP version, but not all platforms use the same middleware.
+
+### Recommendation:
+
+-   write a RIO patch, add to CHIP middleware, upstream to lwip
+    -   Ensure patch is RFC compliant (especially re: expiry)
+
+## DNS
+
+LwIP’s DNS handling isn’t great and breaks down when the router supports
+IPv4/IPv6. There is a single list of DNS servers, DHCP, SLAAC and DHCPv6 all
+update the list without locks. Basically, whatever wrote to the list last gets
+to set the list. Although there is handling for IP type (requesting A or AAAA
+records), there isn’t handling to specify an IPv6 or IPv4 server specifically,
+which can be challenging since not all servers serve all record types.
+
+The design of the weave connectivity manager moves the DNS selection to the
+upper layers by stopping lwip from directly changing the DNS list and hooking to
+the DNS selection. This means the DNS selection policy isn’t hard-coded into the
+lwip layer. This seems like a good model for CHIP going forward.
+
+Additionally, we should ensure that CHIP uses non-blocking DNS APIs.
+
+### Recommendation:
+
+-   bug fix for DHCPv6 to avoid it setting bad addresses.
+    -   note - fixed in
+        https://git.savannah.nongnu.org/cgit/lwip.git/commit/?id=941300c21c45a4dbf1c074b29a9ca3c88c9f6553,
+        but not yet released as a part of an official release.
+-   Create a patch to add hooks to the SetDns and GetDns functions so logic for
+    selecting the DNS server can be moved into the manager layer (possibly
+    already in CHIP, but not present in other middleware)
+
+## Address Scopes
+
+Link local addresses are less common on IPv4, which normally rely on NAT at the
+router to do address translation. CHIP mandates the use of IPv6 link local
+addresses for communication to nodes on the same network (wifi or thread). When
+there is more than one netif in the system (ex. loopback, softAP, STA), the link
+local address needs more information to determine which link the address is
+local to. This is normally added as the link local scope and can be seen on
+addresses ex. FE80::xxxx:xxxx:xxxx:xxxx%<scope>, where the <scope> identifies
+the netif (something like %wlan0 or %eno1 etc.).
+
+Without this indicator, the link local address can only be resolved if there is
+one netif. LwIP will also allow a direct address match to the netif source
+address, but this does not scale well at all and is VERY racy. LwIP also
+supports output to a specific netif, but this is not brought up to the socket
+layer.
+
+Upstream LwIP has support for IPv6 address scopes, but only as an option.
+However, the code to support this is not present in the CHIP LwIP codebase.
+Other platform versions assume this option is not present (ex. M5 has an
+assertion on ip address sizes that disallow the use of a scope tag).
+
+### Recommendation:
+
+-   Update CHIP lwip version for scopes support, ensure CHIP code works with
```

Comment:
Same comment as before for "CHIP lwip version" --> repos does not provide the middleware.